app/api/routers/breeze_buddy/data_sources/handlers.py

```
@@ -0,0 +1,266 @@
+
+
+async def discover_data_source_handler(
+    data_source_id: str,
+    current_user: UserInfo,
+) -> Discovery:
+    """Discover datasets exposed by a data source."""
+    data_source = await get_data_source_handler(data_source_id, current_user)
+    try:
+        adapter = get_adapter(data_source.source_type.value)
+        return await adapter.discover(data_source.config)
+    except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)